libyul/backends/evm/EVMCodeTransform.cpp

```diff
@@ -181,6 +181,7 @@ void CodeTransform::operator()(VariableDeclaration const& _varDecl)
 	{
 		std::visit(*this, *_varDecl.value);
 		expectDeposit(static_cast<int>(numVariables), static_cast<int>(heightAtStart));
+		freeUnusedVariables();
```

Comment:
We should only execute the first part of the `freeUnusedVariables` function because the second part actually pops off the top of the stack. On the other hand, there should be an actual value on the top of the stack here, so it might not be a problem, but it could turn into a problem if we do other modifications to this routine.

Reply:
Yeah, I saw it, but the second half won't be executed, since - due to the visit pushing a slot - none of them will be on the stack top... I could still make it a flag to ``freeUnusedVariables``, though, to make it more obvious that it's used without that here.